front/src/common/Map/Settings/MapSettingsSpeedLimits.tsx

```diff
@@ -1,27 +1,49 @@
 import React, { useEffect, useState } from 'react';
-import PropTypes from 'prop-types';
 import { useSelector, useDispatch } from 'react-redux';
 import { useTranslation } from 'react-i18next';
 import { updateLayersSettings } from 'reducers/map';
 import { IoMdSpeedometer } from 'react-icons/io';
 import SwitchSNCF, { SWITCH_TYPES } from 'common/BootstrapSNCF/SwitchSNCF/SwitchSNCF';
 import SelectImprovedSNCF from 'common/BootstrapSNCF/SelectImprovedSNCF';
 import DotsLoader from 'common/DotsLoader/DotsLoader';
-import { get } from 'common/requests';
 import { setFailure } from 'reducers/main';
 import TIVsSVGFile from 'assets/pictures/layersicons/layer_tivs.svg';
 import { getInfraID } from 'reducers/osrdconf/selectors';
+import { getMap } from 'reducers/map/selectors';
+import { IconType } from 'react-icons';
+import { osrdEditoastApi } from 'common/api/osrdEditoastApi';
 import { FormatSwitch as SimpleFormatSwitch, Icon2SVG } from './MapSettingsLayers';
+import { ApiError } from 'common/api/emptyApi';
+import { SerializedError } from '@reduxjs/toolkit';
 
-function FormatSwitch(props) {
+type FormatSwitchProps = {
+  name: string;
```

Comment:
You can use this, and remove `as keyof typeof layersSettings` lines 97/98
```suggestion
  name:  keyof MapState['layersSettings'];
```

Reply:
Done